### DISCUSSION TITLE
In app messaging

### DISCUSSION THREAD

**Discussion by Septias** [dignifiedquire/dreamer#14] [General]
Hi @dignifiedquire,

I wonder what the best way to send messages between components in egui is. There is no event channel for deeply nested components afaik so a custom channel is needed I guess. 

The most straightforward way atm is just extending the `Command` enum to also handle app communication, but that will get very messy I can imagine. Also, I don't know how to handle the send messages with the correct message receiver. The implementation I made has the flaw that the send command is handled in the communication-loop first and when the rendering code is executed the wanted message is no longer present.

Another option would be to add a second channel for inter-app communication but that would only solve the problem of messiness partly I guess.

What do you think would be the best way to implement that?

**Comment by dignifiedquire** [comment]
Components shouldn not communicate directly with each other in the current design, they just update the main state, and let components react to it.

**Reply by Septias** [reply]
So just add a boolean whether the login-panel should be shown or nah?

**Reply by dignifiedquire** [reply]
yep
